src/CoreWCF.Http/tests/InjectedIncomingPropertiesTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using BasicHttp;
+using CoreWCF.Channels;
+using CoreWCF.Configuration;
+using Helpers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.Http.Tests;
+
+public partial class InjectedIncomingPropertiesTests
+{
+    private readonly ITestOutputHelper _output;
+
+    public InjectedIncomingPropertiesTests(ITestOutputHelper output)
+    {
+        _output = output;
+    }
+
+    [Fact]
+    public void BasicHttpRequestReplyEchoString()
+    {
+        string testString = new('a', 3000);
+        IWebHost host = ServiceHelper.CreateWebHostBuilder<Startup>(_output).Build();
+        using (host)
+        {
+            host.Start();
+            System.ServiceModel.BasicHttpBinding httpBinding = ClientHelper.GetBufferedModeBinding();
+            var factory = new System.ServiceModel.ChannelFactory<ClientContract.IEchoService>(httpBinding,
+                new System.ServiceModel.EndpointAddress(new Uri($"http://localhost:{host.GetHttpPort()}/BasicWcfService/basichttp.svc")));
+            ClientContract.IEchoService channel = factory.CreateChannel();
+            string result = channel.EchoString(testString);
+            Assert.Equal(testString, result);
+        }
+    }
+
+    internal class Startup
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddServiceModelServices();
+        }
+
+        public void Configure(IApplicationBuilder app)
+        {
+            app.UseServiceModel(builder =>
+            {
+                builder.AddService<Services.EchoService>();
+                builder.AddServiceEndpoint<Services.EchoService, ServiceContract.IEchoService>(new CoreWCF.BasicHttpBinding(), "/BasicWcfService/basichttp.svc");
+            });
```

Comment:
Should this test be using InjectedIncomingPropertiesService/IInjectedIncomingPropertiesService instead of EchoService/IEchoService?

Reply:
Good catch thanks